lib/modules/datasource/go/base.ts

```diff
@@ -154,17 +155,35 @@ export class BaseGoDatasource {
       const parsedUrl = URL.parse(goSourceUrl);
 
       // TODO: `parsedUrl.pathname` can be undefined
-      const packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
+      let packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
 
-      const registryUrl = `${parsedUrl.protocol}//${parsedUrl.host}`;
+      const endpoint = GlobalConfig.get('endpoint');
+
+      const endpointPrefix = endpoint?.match('https://[^/]*/(.*?/)(api/v4/?)?');
+
+      if (endpointPrefix) {
+        packageName = packageName.replace(endpointPrefix[1], '');
+      }
+
+      const registryUrl =
+        endpoint?.replaceAll('api/v4/', '') ??
```

Comment:
```suggestion
        endpoint?.replace(regEx('api/v4/?$'), '') ??
```

Should only occure once on the end.
I think the `registryUrl` should only be set to endpoint if `endpointPrefix` was matched.